Src/FluentAssertions/Types/TypeAssertions.cs

```diff
@@ -72,6 +72,45 @@ public new AndConstraint<TypeAssertions> BeAssignableTo<T>(string because = "",
             return BeAssignableTo(typeof(T), because, becauseArgs);
         }
 
+        bool CheckIfAssignableToOpenGeneric(Type definition)
```

Comment:
🔧 Splitting the method in two, one for interfaces and one for classes might help read this code
🔧 Please move the method below the first use. We follow the order of execution ordering rule.